tf_explain/core/vanilla_gradients.py

```diff
@@ -56,19 +123,13 @@ def compute_gradients(images, model, class_index):
         Returns:
             tf.Tensor: 4D-Tensor
         """
-        num_classes = model.output.shape[1]
-
-        expected_output = tf.one_hot([class_index] * images.shape[0], num_classes)
-
         with tf.GradientTape() as tape:
             inputs = tf.cast(images, tf.float32)
             tape.watch(inputs)
-            predictions = model(inputs)
-            loss = tf.keras.losses.categorical_crossentropy(
-                expected_output, predictions
-            )
+            scores = model(inputs)
+            scores_for_class = scores[:, class_index : class_index + 1]
```

Comment:
Why `scores[:, class_index : class_index + 1]` and not just `scores[:, class_index]` ?